dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/compute/functions/OsImageToImage.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.dimensiondata.cloudcontrol.compute.functions;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Function;
+import com.google.common.base.Splitter;
+import com.google.common.base.Supplier;
+import com.google.common.collect.FluentIterable;
+import com.google.inject.Inject;
+import org.jclouds.collect.Memoized;
+import org.jclouds.compute.domain.Image;
+import org.jclouds.compute.domain.ImageBuilder;
+import org.jclouds.compute.domain.OperatingSystem;
+import org.jclouds.compute.domain.OsFamily;
+import org.jclouds.dimensiondata.cloudcontrol.domain.OsImage;
+import org.jclouds.domain.Location;
+import org.jclouds.location.predicates.LocationPredicates;
+
+import javax.inject.Singleton;
+import java.util.Set;
+
+import static com.google.common.collect.Iterables.get;
+
+@Singleton
+public class OsImageToImage implements Function<OsImage, Image> {
+
+   private final Supplier<Set<Location>> locations;
+   private final Function<String, OsFamily> imageDescriptionToOsFamily;
+
+   @Inject
+   OsImageToImage(@Memoized final Supplier<Set<Location>> locations,
+         Function<String, OsFamily> imageDescriptionToOsFamily) {
+      this.locations = locations;
+      this.imageDescriptionToOsFamily = imageDescriptionToOsFamily;
+   }
+
+   @Override
+   public Image apply(OsImage input) {
+
+      OsFamily osFamily = imageDescriptionToOsFamily.apply(input.description());
+      String osVersion = parseVersion(input.description());
+
+      OperatingSystem os = OperatingSystem.builder().name(input.name()).description(input.description())
+            .family(osFamily).version(osVersion).is64Bit(is64bit(input)).build();
+
+      return new ImageBuilder().id(input.id()).name(input.name()).description(input.description())
+            .status(Image.Status.AVAILABLE).operatingSystem(os).location(
+                  FluentIterable.from(locations.get()).firstMatch(LocationPredicates.idEquals(input.datacenterId()))
```

Comment:
Just one consideration about the meaning of the `location` field in an image:

* If the image is globally available in all locations (the same image, same id, etc), then the location field should be `null`.
* if the image is only available in certain locations, then the location field must be set accordingly.

Reply:
Image is only available in certain locations, so what I have here is valid.